Zend/zend_objects.c

```diff
@@ -282,3 +289,27 @@ ZEND_API zend_object *zend_objects_clone_obj(zend_object *old_object)
 
 	return new_object;
 }
+
+ZEND_API zend_object *zend_objects_new_persistent(zend_class_entry *ce)
+{
+	size_t alloc_size = sizeof(zend_object) + zend_object_properties_size(ce);
+	zend_object *obj = malloc(alloc_size);
+	_zend_object_minimal_init(obj, ce);
+	GC_TYPE_INFO(obj) |= IS_OBJ_PERSISTENT << GC_FLAGS_SHIFT;
+	obj->handle = zend_objects_persistent_handle_counter;
+	zend_objects_persistent_handle_counter += (alloc_size - 1) / sizeof(zend_object) + 1;
```

Comment:
If you're doing something like this anyway, why is persistent_objects/zend_objects_persistent_handle_counter quantized to zend_objects rather than just a raw memory region?

Reply:
Probably does not make too much sense, the main reason is that I'm accessing `EG(persistent_objects) + obj->handle` where EG(persistent_objects) is a zend_object-array. But could just cast to char and back to zend_object at that point.

Reply:
I mean, why is it `zend_object*` in the first place, rather than `char*`?

Reply:
Because it contains objects :-D that was the logical thought I had, but obviously, this chunk of memory can just be made more generic and the zend_objects_persistent_handle_counter just be a counter. There's no reason why this chunk of memory cannot be just shared between multiple potential usages. (and then I'd also decouple that "handle_counter" from zend_objects.c)